extide/gradle/src/org/netbeans/modules/gradle/api/execute/RunUtils.java

```diff
@@ -539,6 +541,39 @@ public static ReplaceTokenProvider simpleReplaceTokenProvider(final String token
             }
         };
     }
+    
+    /**
+     * Returns an action mapping provider for the specified project and context. Specifically 
+     * supports {@link ProjectConfiguration}s, so the returned provider will 
+     * @param p the project
+     * @param context the action invocation context
+     * @return action provider suitable for the project/context.
+     * @since 2.14
+     */
+    public static ProjectActionMappingProvider findActionProvider(Project p, Lookup context) {
+        ConfigurableActionProvider cap = p.getLookup().lookup(ConfigurableActionProvider.class);
+        if (cap == null) {
+            return p.getLookup().lookup(ProjectActionMappingProvider.class);
+        }
+        GradleExecConfiguration cfg = ProjectConfigurationSupport.getEffectiveConfiguration(p, context);
+        return cap.findActionProvider(cfg.getId());
+    }
+
+    /**
+     * Checks if the action is disabled. Use in preference to plain <code>actionMapping == null</code>
+     * as it also handles a mapping that forcefully deconfigures an action.
+     * 
+     * @param am action mapping to check
+     * @return true, if the action is <b>disabled</b>
+     * @since 2.14
+     */
+    public static boolean isActionDisabled(ActionMapping am) {
```

Comment:
Would be better as a static method in ActionMapping.isDisabled(ActionMapping)

Reply:
OK :) to be more fun, I had it that way (both isDisabled and DEFAULT) originally, then I've decided not to stretch the new Java features that much and refactored. No real preference from my side.

Reply:
Done in 688e72917b